store/sqlstore/channel_store.go

```diff
@@ -1041,19 +1041,23 @@ func (s SqlChannelStore) GetChannels(teamId string, userId string, opts *model.C
 			sq.Eq{"ch.TeamId": ""},
 		})
 	}
-
-	if opts.IncludeDeleted {
-		if opts.LastDeleteAt != 0 {
-			// We filter by non-archived, and archived >= a timestamp.
-			query = query.Where(sq.Or{
-				sq.Eq{"ch.DeleteAt": 0},
-				sq.GtOrEq{"ch.DeleteAt": opts.LastDeleteAt},
-			})
-		}
-		// If opts.LastDeleteAt is not set, we include everything. That means no filter is needed.
+	if opts.Deleted {
+		// this will be false unless specified
+		query = query.Where(sq.NotEq{"ch.DeleteAt": 0})
 	} else {
-		// Don't include archived channels.
-		query = query.Where(sq.Eq{"ch.DeleteAt": 0})
+		if opts.IncludeDeleted {
+			if opts.LastDeleteAt != 0 {
+				// We filter by non-archived, and archived >= a timestamp.
+				query = query.Where(sq.Or{
+					sq.Eq{"ch.DeleteAt": 0},
+					sq.GtOrEq{"ch.DeleteAt": opts.LastDeleteAt},
+				})
+			}
+			// If opts.LastDeleteAt is not set, we include everything. That means no filter is needed.
```

Comment:
I know this wasn't your comment, but I think it's more accurate to say this since LastDeleteAt is always set.

```suggestion
			// The default zero value of opts.LastDeleteAt results in all deleted channels being included.
```

Reply:
I've opened up a PR to also document the option fields themselves: https://github.com/mattermost/mattermost-server/pull/22009